parts/windows/kuberneteswindowsfunctions.ps1

```diff
@@ -214,6 +214,17 @@ function Register-NodeResetScriptTask {
     Register-ScheduledTask -TaskName "k8s-restart-job" -InputObject $definition
 }
 
+function Register-NodeLabelSyncScriptTask {
+    Write-Log "Creating a periodical task to run windowsnodelabelsync.ps1"
+
+    $action = New-ScheduledTaskAction -Execute "powershell.exe" -Argument "-File `"c:\k\windowsnodelabelsync.ps1`""
+    $principal = New-ScheduledTaskPrincipal -UserId SYSTEM -LogonType ServiceAccount -RunLevel Highest
+    # trigger this task once(by `-Once) at the time being scheduled(by `-At (Get-Date).Date`) every minute(by `-RepetitionInterval 00:01:00`)
+    $trigger = New-ScheduledTaskTrigger -Once -At (Get-Date).Date -RepetitionInterval 00:01:00

```

Comment:
https://docs.microsoft.com/en-us/powershell/module/scheduledtasks/new-scheduledtasktrigger?view=win10-ps
It may not work as the expected per the doc for `-ThrottleLimit`. I suggest to test it in windows agent node.
```
Specifies the maximum number of concurrent operations that can be established to run the cmdlet. If this parameter is omitted or a value of 0 is entered, then Windows PowerShell® calculates an optimum throttle limit for the cmdlet based on the number of CIM cmdlets that are running on the computer. The throttle limit applies only to the current cmdlet, not to the session or to the computer.
```

Reply:
Just check the usage in windows node pool, it works fine. BTW, `New-ScheduledTaskTrigger` was the one I used to test on aks-engine.
```
PS C:\>  Get-ScheduledTaskInfo -TaskName "sync-kubelet-node-label"


LastRunTime        : 1/21/2021 8:28:28 AM
LastTaskResult     : 0
NextRunTime        : 1/21/2021 8:29:29 AM
NumberOfMissedRuns : 0
TaskName           : sync-kubelet-node-label
TaskPath           :
PSComputerName     :
```
`kubernetes.azure.com/role--changekey=agent,storageprofile--changekey=managed--changevalue,storageprofile-changekey=managed-changevalue` This is the changed result, this is made by changing the nodelabel entry in `c:\k\kubeclusterconfig.json`
```
PS C:\> c:\k\kubectl.exe --kubeconfig=c:\k\config  get node akswinvn000000 --show-labels        
NAME             STATUS   ROLES   AGE   VERSION    LABELS
akswinvn000000   Ready    agent   31d   v1.18.10   abc=def,agentpool=winvn,beta.kubernetes.io/ar
ch=amd64,beta.kubernetes.io/instance-type=Standard_D2s_v3,beta.kubernetes.io/os=windows,failure-
domain.beta.kubernetes.io/region=westus2,failure-domain.beta.kubernetes.io/zone=0,kubernetes.azu
re.com/cluster=MC_qinhao-win-calico_calico-win_westus2,kubernetes.azure.com/node-image-version=A
KSWindows-2019-17763.1577.201111,kubernetes.azure.com/role--changekey=agent,kubernetes.azure.com
/role-changekey=agent,kubernetes.azure.com/role=agent,kubernetes.io/arch=amd64,kubernetes.io/hos
tname=akswinvn000000,kubernetes.io/os=windows,kubernetes.io/role=agent,node-role.kubernetes.io/a
gent=,node.kubernetes.io/instance-type=Standard_D2s_v3,node.kubernetes.io/windows-build=10.0.177
63,storageprofile--changekey=managed--changevalue,storageprofile-changekey=managed-changevalue,s
torageprofile=managed,storagetier=Premium_LRS,topology.kubernetes.io/region=westus2,topology.kub
ernetes.io/zone=0
```

Reply:
Per the doc, Windows Powershell will calculate an optimum throttle limit. I do not know whether it will limit the scheduled task since this task is ran every one minute.

Reply:
BTW: I cannot find any limitation in `New-JobTrigger`. I think that it is better if we can use `New-JobTrigger` considering the unknown `optimum throttle limit`

Reply:
I cannot create a  new trigger through `New-JobTrigger`  after several tries.  Do you have any idea?
```
PS C:\> New-JobTrigger -AtStartup -RandomDelay 00:00:05 -RepetitionInterval 00:01:00 -RepeatInde
finitely
New-JobTrigger : Parameter set cannot be resolved using the specified named parameters. 
At line:1 char:1
+ New-JobTrigger -AtStartup -RandomDelay 00:00:05 -RepetitionInterval 0 ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [New-JobTrigger], ParameterBindingException  
    + FullyQualifiedErrorId : AmbiguousParameterSet,Microsoft.PowerShell.ScheduledJob.NewJobTr  
   iggerCommand
```

Reply:
Changed to use `New-JobTrigger` instead. Thanks for pointing this out.